src/npctalk.cpp

```diff
@@ -2364,7 +2368,11 @@ void talk_effect_fun_t::set_weighted_list_eocs( const JsonObject &jo,
         }
         eocs.add( eoc, weight );
     }
-    function = [&eocs]( const dialogue & ) {
+    function = [eocs]( const dialogue & ) {
```

Comment:
No need to copy it
```suggestion
    function = [&eocs]( const dialogue & ) {
```

Reply:
In the portal branch if you don't copy it you get an invalid array error.  Not an expert on c++ memory handling but pretty sure if you don't copy it, it will go out of scope if set for the future.

Reply:
Oh, you're setting a member variable to this function. Yeah, eocs will go out of scope and then this function will trigger UAF.